python/IECoreMaya/FnSceneShape.py

```diff
@@ -409,74 +409,192 @@ def convertAllToGeometry( self ) :
 
 			# turn the scene node an intermediateObject so it can't be seen by LiveScene
 			fn.findPlug( "intermediateObject" ).setBool( True )
-	
-	## Converts the object (if any) in the scene interface into maya geometry.
-	# If a shape with the expected name but incompatible type is found under the transform, we rename it and create a new proper shape.
-	# The shape is connected to the scene shape object output only if it isn't already connected or locked.
-	# transformNode parameter can be used to specify the parent of the geometry. If None, uses the transform of the scene shape.
-	def convertObjectToGeometry( self, transformNode = None ):
-		
-		if not self.sceneInterface().hasObject():
-			return
 
-		if not transformNode:
-			# No transform provided, use the transform of the reader
-			dag = maya.OpenMaya.MDagPath()
-			self.getPath( dag )
-			dag.pop()
-			transformNode = dag.fullPathName()
+	## Returns a dict which keys are paramter names and values are the parameter values. Value types are always str.
+	__convertParamExp = re.compile( "-([a-zA-Z_]\w*)\s(\w+)" )
```

Comment:
For performance I used regexp. It not exactly the same as how SceneShapeInterface interprets the string but I think the difference is practically subtle.